modules/core/src/test/java/org/apache/synapse/message/store/impl/jms/JmsMessageStoreTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.synapse.message.store.impl.jms;
+
+import junit.framework.Assert;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.axiom.soap.impl.llom.soap11.SOAP11Factory;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.core.axis2.Axis2SynapseEnvironment;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.apache.activemq.broker.BrokerService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Test class for Jms Store implementation
+ */
+public class JmsMessageStoreTest {
```

Comment:
shouldn't we say this activeMQ jms test case or something and add a test case for wso2 mb as well?

Reply:
In this test case, we're using an embedded activemq broker. The broker is not running as a separate process. But with wso2 mb, we have to start the broker separately. Anyhow we can try that with andes. But this test case tests only the jms store implementation.

Reply:
@Ramith, by default jms test cases are written to run with activeMQ. We haven't named the test cases as such in any of the existing test cases.
@ThishaniLucas if we are to add a test case for wso2mb for this tests-platform -> tests-wso2mb is the place for that.